### PR TITLE
fix(cardutilityfunctions): add null handler

### LINKED ISSUE #3725: [cardUtiltiyFunctions] findMatchingThresholds throws error when item is null
### What package is this for?

- [ X] React
- [ ] Angular

### Describe the bug

Hotspot content breaks app when configured with thresholds but no readings provided. findMatchingThresholds method does not handle item value as _null_

### To Reproduce

1. Go to Dashboard editor
2. Create an image card and add image to it
3. Edit image to include a hotspot anywhere in the image
4. Give it a name, description and icon. 
5. Select a datasource to associate the hotspot and edit the data item to add a threshold to it.
6. Save dashboard
7. Find the dashboard in monitor and click in the hotspot icon
8. See error

> Please create a reduced test case in CodeSandbox:
> https://codesandbox.io/s/github/carbon-design-system/carbon-addons-iot-react/tree/next/packages/react/examples/codesandbox

### Expected behavior

It expected to see the popup with datasource and no data (e.g. humidity: --).

### Environment/versions:

- OS: MacOS
- Browser: chrome
- `carbon-addons-iot-react` version: `v2.152.0`

### Specific timeline issues / requests

Do you want this work within a specific time period? Is it related to an
upcoming release?

It can be assign to myself

_NB: The core contributors will try to work with your timeline, but it's not
guaranteed. The earlier you make a request in advance of a desired delivery
date, the better!_

### PR DESCRIPTION
**Pull request by vaffonso**
Closes #3725

**Summary**

- Add handler to prevent break for hotspot content with threshold and no readings

**Change List (commits, features, bugs, etc)**

- cardUtilityFunctions: Add handler to null input for findMatchingThresholds method
- test-app.yml: set node version due to [build error](https://github.com/carbon-design-system/carbon-addons-iot-react/actions/runs/4224697760/jobs/7336233752)

**Acceptance Test (how to verify the PR)**

1. Go to Dashboard editor
2. Create an image card and add image to it
3. Edit image to include a hotspot anywhere in the image
4. Give it a name, description and icon.
5. Select a datasource to associate the hotspot and edit the data item to add a threshold to it.
6. Save dashboard
7. Find the dashboard in monitor and click in the hotspot icon

**Regression Test (how to make sure this PR doesn't break old functionality)**

- tests here

**Things to look for during review**

- [ ] Make sure all references to `iot` or `bx` class prefix is using the prefix variable
- [ ] (React) All major areas have a `data-testid` attribute. New test ids should have test written to ensure they are not changed or removed.
- [ ] UI should be checked in RTL mode to ensure the proper handling of layout and text.
- [ ] All strings should be translatable.
- [ ] The code should pass a11y scans (The storybook a11y knob should show no violations). New components should have a11y test written.
- [ ] Unit test should be written and should have a coverage of 90% or higher in all areas.
- [ ] All components should be passing visual regression test. For new styles or components either a visual regression test should be written for all permutations or the base image updated.
- [ ] Changes or new components should either write new or update existing documentation.
- [ ] PR should link and close out an existing issue

### CODE REVIEW

**Comment by vaffonso** [comment]
@EdmondCShaffer @herleraja can you take a look?

**Review by EdmondCShaffer — APPROVED** [review]

**Comment by vaffonso** [comment]
I updated test-app.yml due to this build error
https://github.com/carbon-design-system/carbon-addons-iot-react/actions/runs/4224697760/jobs/7336233752

**Comment by erzhan-temir-mamyrov** [comment]
LGTM 👍

File: .github/workflows/test-app.yml [hunk 1]
```diff
@@ -77,6 +77,12 @@ jobs:
           filters: |
             react:
               - 'packages/react/**'
+      
+      - name: Use Node.js
+        uses: actions/setup-node@v1
+        if: steps.changes.outputs.react == 'true'
+        with:
+          node-version: '16.x'
```
Comment (davidicus) [inline .github/workflows/test-app.yml:85]: This change has already been made. This was failing across all PRs and a fix was delivered last week.

Comment (davidicus) [reply .github/workflows/test-app.yml:85]: it looks identical I would not think its a problem

**Review by davidicus — APPROVED** [review]

**Pull request merged**